File: src/main/java/um/haberes/rest/model/CargoClaseDetalle.java
/**
 * 
 */
package um.haberes.rest.model;

import java.io.Serializable;
import java.math.BigDecimal;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.OneToOne;
import jakarta.persistence.Table;
import jakarta.persistence.UniqueConstraint;

import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

/**
 * @author daniel
 *
 */
@Entity
@Table(uniqueConstraints = { @UniqueConstraint(columnNames = { "cargoClasePeriodoId", "anho", "mes" }) })
@EqualsAndHashCode(callSuper = false)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class CargoClaseDetalle extends Auditable implements Serializable {
	/**
	 * 
	 */
	private static final long serialVersionUID = 1538133904656877507L;

	@Id
	@GeneratedValue(strategy = GenerationType.IDENTITY)
	private Long cargoClaseDetalleId;

	private Long legajoId;
	private Integer anho = 0;
	private Integer mes = 0;
	private Long cargoClaseId;
	private Integer dependenciaId;
	private Integer facultadId;
	private Integer geograficaId;
	private Integer horas = 0;
	private BigDecimal valorHora = BigDecimal.ZERO;
	private Long cargoClasePeriodoId;
	private Byte liquidado = 0;

	@OneToOne
	@JoinColumn(name = "legajoId", insertable = false, updatable = false)
	private Persona persona;

	@OneToOne
	@JoinColumn(name = "cargoClaseId", insertable = false, updatable = false)
	private CargoClase cargoClase;

	@OneToOne
	@JoinColumn(name = "dependenciaId", insertable = false, updatable = false)
	private Dependencia dependencia;

	@OneToOne
	@JoinColumn(name = "facultadId", insertable = false, updatable = false)
	private Facultad facultad;

	@OneToOne
	@JoinColumn(name = "geograficaId", insertable = false, updatable = false)
	private Geografica geografica;

	@OneToOne
	@JoinColumn(name = "cargoClasePeriodoId", insertable = false, updatable = false)
	private CargoClasePeriodo cargoClasePeriodo;

}

File: src/main/java/um/haberes/rest/kotlin/model/CargoClaseDetalle.kt
package um.haberes.rest.kotlin.model

import jakarta.persistence.*
import java.math.BigDecimal

@Entity
@Table(uniqueConstraints = [UniqueConstraint(columnNames = ["cargoClasePeriodoId", "anho", "mes"])])
data class CargoClaseDetalle(

    @Id @GeneratedValue(strategy = GenerationType.IDENTITY)
    var cargoClaseDetalleId: Long? = null,

    var legajoId: Long? = null,
    var anho: Int? = 0,
    var mes: Int? = 0,
    var cargoClaseId: Long? = null,
    var dependenciaId: Int? = null,
    var facultadId: Int? = null,
    var geograficaId: Int? = null,
    var horas: Int = 0,
    var valorHora: BigDecimal = BigDecimal.ZERO,
    var cargoClasePeriodoId: Long? = null,
    var liquidado: Byte = 0,

    @OneToOne
    @JoinColumn(name = "legajoId", insertable = false, updatable = false)
    var persona: Persona? = null,

    @OneToOne
    @JoinColumn(name = "cargoClaseId", insertable = false, updatable = false)
    var cargoClase: CargoClase? = null,

    @OneToOne
    @JoinColumn(name = "dependenciaId", insertable = false, updatable = false)
    var dependencia: Dependencia? = null,

    @OneToOne
    @JoinColumn(name = "facultadId", insertable = false, updatable = false)
    var facultad: Facultad? = null,

    @OneToOne
    @JoinColumn(name = "geograficaId", insertable = false, updatable = false)
    var geografica: Geografica? = null,

    @OneToOne
    @JoinColumn(name = "cargoClasePeriodoId", insertable = false, updatable = false)
    var cargoClasePeriodo: CargoClasePeriodo? = null


) : Auditable()
